syntax = "proto3";

package document_processor;

service DocumentProcessor {
  rpc ProcessDocument (ProcessDocumentRequest) returns (ProcessDocumentResponse);
  rpc ModifyDocument (ModifyDocumentRequest) returns (ModifyDocumentResponse);
}

message Segment{
  uint32 segment_id = 1;
  string original_text = 2;
  string translated_text = 3;
}

message ProcessDocumentRequest {
  string content = 1;
  string file_type = 2;
}

message ProcessDocumentResponse {
  repeated  Segment segments = 1;
}



message ModifyDocumentRequest {
  string content = 1;
  string replacements = 2;
  string file_type = 3;
}

message ModifyDocumentResponse {
  string modified_content = 1;
}
